Text/HealthcareTA/HealthcareTA.cs

```diff
@@ -0,0 +1,136 @@
+                async (inRecord, outRecord) => {
+                    // Prepare analysis operation input
+                    var document = inRecord.Data["document"] as string;
+                    var options = new AnalyzeHealthcareEntitiesOptions { };
+                    if (document.Length >= maxCharLength)
+                    {
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = $"Healthcare Text Analytics Error: The submitted document was over {maxCharLength} characters. It has been truncated to fit this requirement."
+                        });
+                        document = document.Substring(0, maxCharLength);
+                    }
+                    List<string> batchInput = new List<string>()
+                    {
+                        document
+                    };
+
+                    // start analysis process
+                    var timer = System.Diagnostics.Stopwatch.StartNew();
+                    AnalyzeHealthcareEntitiesOperation healthOperation = await client.StartAnalyzeHealthcareEntitiesAsync(batchInput, "en", options);
+                    var healthOperationTask = healthOperation.WaitForCompletionAsync().AsTask();
+
+                    if (await Task.WhenAny(healthOperationTask, Task.Delay(timeoutMiliseconds)) == healthOperationTask)
+                    {
+                        // Task Completed, now lets process the result.
+                        outRecord.Data["status"] = healthOperation.Status.ToString();
+                        if (healthOperation.Status != TextAnalyticsOperationStatus.Succeeded)
+                        {
+                            // The operation was not a success
+                            outRecord.Warnings.Add(new WebApiErrorWarningContract
+                            {
+                                Message = "Healthcare Text Analytics Error: Health Operation returned a non-succeeded status."
+                            });
+                        }
+                        else
+                        {
+                            // The operation was a success, so lets add the results to our output.
+                            await ExtractEntityData(healthOperation.Value, outRecord);
+                        }
+                    }
+                    else
+                    {
+                        // Timeout
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = "Healthcare Text Analytics Error: The Text Analysis Operation took too long to complete."
+                        });
+                    }
+
+                    // Record how long this task took to complete.
+                    timer.Stop();
+                    outRecord.Data["timeToComplete"] = timer.Elapsed.TotalSeconds.ToString();
```

Comment:
I don't think we'll actually want to check this in since customers likely won't care. We can add a log for it in the Azure Function for those that do, it just doesn't need to be an output if that makes sense.